client/src/components/Graph/D3Plot.vue

```diff
@@ -0,0 +1,307 @@
+        // The different color options are here: https://github.com/d3/d3-scale-chromatic
+        .interpolator(d3.interpolateCool);
+
+      const legendWidth = this.generateLegend(svg, colorGen);
+
+      const leftPadding = yLabelWidth + 45;
+      const bottomPadding = xLabelHeight + 30;
+      const rightPadding = legendWidth;
+
+      const minXVal = d3.min(datasets, (d) => d3.min(d.xs));
+      const minYVal = d3.min(datasets, (d) => d3.min(d.ys));
+      const maxXVal = d3.max(datasets, (d) => d3.max(d.xs));
+      const maxYVal = d3.max(datasets, (d) => d3.max(d.ys));
+
+      let xScale;
+      let yScale;
+      if (this.d3PlotData.plot_details.yscale === 'log') {
+        xScale = d3.scaleLog();
+        yScale = d3.scaleLog();
+      } else {
+        xScale = d3.scaleLinear();
+        yScale = d3.scaleLinear();
+      }
+      xScale = xScale
+        .domain([minXVal, maxXVal])
+        .range([leftPadding, w - rightPadding]);
+      yScale = yScale
+        .domain([minYVal, maxYVal])
+        .range([h - bottomPadding, bottomPadding]);
+
+      // Take each set of data points and put them on the plot
+      Object.values(datasets).forEach((dataset, i) => {
+        // Map each point to an array where it has the x, y value in each entry
+        const coordsArr = dataset.xs.map((val, index) => [
+          val,
+          dataset.ys[index],
+        ]);
+        svg.append('path')
+          .datum(coordsArr)
+          .attr('fill', 'none')
+          .attr('class', `line-${i}`)
+          .attr('stroke', colorGen(i))
+          .attr('stroke-width', 1.5)
+          .attr('d', d3.line()
+            .x((data) => xScale(data[0]))
+            .y((data) => yScale(data[1])));
+        svg.append('path');
+      });
+
+      const xAxis = d3.axisBottom(xScale);
+      const yAxis = d3.axisLeft(yScale);
+
+      svg.append('g')
+        .attr('id', 'x-axis')
+        .attr(
+          'transform',
+          `translate(0,${h - bottomPadding})`,
+        )
+        .call(xAxis);
+      svg.append('g')
+        .attr('id', 'y-axis')
+        .attr('transform', `translate(${leftPadding},0)`)
+        .call(yAxis);
+
+      // gridlines in x axis function
+      function make_x_gridlines() {
+        return d3.axisBottom(xScale)
+          .ticks(6);
+      }
+
+      // gridlines in y axis function
+      function make_y_gridlines() {
+        return d3.axisLeft(yScale)
+          .ticks(6);
+      }
+
+      // add the X gridlines
+      svg.append('g')
+        .attr('class', 'grid')
+        .attr('transform', `translate(0,${h - bottomPadding})`)
+        .call(make_x_gridlines()
+          .tickSize(-(h - bottomPadding * 2))
+          .tickFormat(''));
+
+      // add the Y gridlines
+      svg.append('g')
+        .attr('class', 'grid')
+        .attr('transform', `translate(${leftPadding},0)`)
+        .call(make_y_gridlines()
+          .tickSize(-(w - rightPadding - leftPadding))
+          .tickFormat(''));
+    },
```

Comment:
Can this logic be moved to another file since its just JS and doesn't use Vue functionality?

Reply:
It does use Vue functionality in that it is called whenever the data changes from the store. You are right it could be moved to a JS file separately, with an argument which specifies the id of the element to modify, although what would the file be called and where should it go? All of this code is very much having to do with the interface, so it doesn't seem like it would belong in the `utils` folder. Up to you.

Reply:
It uses Vue functionality that could just as easily be one line  (watch: READ_ONLY.plotdata{buildPlot();) in the parent component. And yes its related to the interface but none of these functions interact with the components state, props, or any other functionality related to it actually being in a component. They take data as an input then build a javascript object and tie that to an html element with a provided id. 

And I'd say a function like buildPlot would very much qualify as a utility. Maybe utils/plot.js?

Reply:
This was done as requested.